plugins/UM3NetworkPrinting/resources/qml/CameraButton.qml

```diff
@@ -0,0 +1,47 @@
+import QtQuick 2.3
+import QtQuick.Controls 1.4
+import QtQuick.Controls.Styles 1.3
+import QtQuick.Controls 2.0 as Controls2
+import QtGraphicalEffects 1.0
+
+import UM 1.3 as UM
+import Cura 1.0 as Cura
+
+Rectangle
```

Comment:
I'm not sure if I initially built it like that, but this should just be a button. You can set the label of a button to be an image and the background to be an item. That will do what you want with a fraction of the code (and probably with more optimisations possible for the QML engine)

Reply:
Nope. You initially built it twice in two places so this was already an optimization. It took a little too much extra work to make it truly reusable since we need to make a reusable button component for Cura 4.0 anyway. This was just a quick little refactor.